pkg/asset/agent/image/ignitionbase.go

```diff
@@ -0,0 +1,269 @@
+	// registryCABundle := &mirror.CaBundle{}
+	// dependencies.Get(registriesConfig, registryCABundle)
+
+	publicContainerRegistries := getPublicContainerRegistries(registriesConfig)
+
+	logrus.Infof("The release image is %s", clusterImageSet.Spec.ReleaseImage)
+	releaseImageMirror := mirror.GetMirrorFromRelease(clusterImageSet.Spec.ReleaseImage, registriesConfig)
+
+	infraEnvID := uuid.New().String()
+	logrus.Debug("Generated random infra-env id ", infraEnvID)
+	a.infraEnvID = infraEnvID
+
+	osImage, err := getOSImagesInfo(archName)
+	if err != nil {
+		return err
+	}
+	a.osImage = *osImage
+	a.CPUArch = *osImage.CPUArchitecture
+
+	agentTemplateData := &agentTemplateData{
+		PullSecret:                pullSecretAsset.GetPullSecretData(),
+		ReleaseImages:             releaseImageList,
+		ReleaseImage:              clusterImageSet.Spec.ReleaseImage,
+		ReleaseImageMirror:        releaseImageMirror,
+		HaveMirrorConfig:          len(registriesConfig.MirrorConfig) > 0,
+		PublicContainerRegistries: publicContainerRegistries,
+		InfraEnvID:                a.infraEnvID,
+		OSImage:                   a.osImage,
+		Proxy:                     infraEnv.Spec.Proxy,
+	}
+
+	err = bootstrap.AddStorageFiles(&config, "/", "agent/files", agentTemplateData)
+	if err != nil {
+		return err
+	}
+
+	// Set up bootstrap service recording
+	if err := bootstrap.AddStorageFiles(&config,
+		"/usr/local/bin/bootstrap-service-record.sh",
+		"bootstrap/files/usr/local/bin/bootstrap-service-record.sh",
+		nil); err != nil {
+		return err
+	}
+
+	// Use bootstrap script to get container images
+	relImgData := struct{ ReleaseImage string }{
+		ReleaseImage: clusterImageSet.Spec.ReleaseImage,
+	}
+	for _, script := range []string{"release-image.sh", "release-image-download.sh"} {
+		if err := bootstrap.AddStorageFiles(&config,
+			"/usr/local/bin/"+script,
+			"bootstrap/files/usr/local/bin/"+script+".template",
+			relImgData); err != nil {
+			return err
+		}
+	}
+
+	filesToInclude := [...]asset.File{
+		*infraEnvAsset.File,
+		*clusterImageSetAsset.File,
+		*pullSecretAsset.File,
+	}
+
+	for _, file := range filesToInclude {
+		manifestFile := ignition.FileFromBytes(filepath.Join(manifestPath, filepath.Base(file.Filename)),
+			"root", 0600, file.Data)
+		config.Storage.Files = append(config.Storage.Files, manifestFile)
+	}
+
+	disabledServices := []string{"set-hostname.service", "start-cluster-installation.service"}
+	enabledServices := StringArrayRemove(agentEnabledServices, disabledServices)
+	err = bootstrap.AddSystemdUnits(&config, "agent/systemd/units", agentTemplateData, enabledServices)
+	if err != nil {
+		return err
+	}
+
+	a.Config = config
+
+	if err := a.generateFile(unconfiguredIgnitionFilename); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// PersistToFile writes the unconfigured ignition in the assets folder.
+func (a *IgnitionBase) PersistToFile(directory string) error {
+	unconfiguredIgnFile := filepath.Join(directory, a.File.Filename)
```

Comment:
I'm getting "nil pointer dereference" here when running 'openshift-install agent create unconfigured-ignition'.
I guess 'a.File' is missing?

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3a325df]

goroutine 1 [running]:
github.com/openshift/installer/pkg/asset/agent/image.(*IgnitionBase).PersistToFile(0x229cd500, {0x7ffedeade49b?, 0x229cd500?})
	/home/git/installer/pkg/asset/agent/image/ignitionbase.go:183 +0x3f
main.runTargetCmd.func1({0x7ffedeade49b, 0xb})
	/home/git/installer/cmd/openshift-install/create.go:265 +0x24a
main.runTargetCmd.func2(0x229a1da0?, {0xc00059b020?, 0x2?, 0x2?})
	/home/git/installer/cmd/openshift-install/create.go:290 +0xe7
```

Reply:
Yes a.File is missing. This could happen if one of the assets that IgnitionBase depends on is missing. I will add a check and additional integration test validations to check for these cases.